<?xml version="1.0" encoding="UTF-8"?>
<launch>


	<!-- <arg name =  "drone_name" default = "drone_1" /> -->
	<arg name = "pose_topic"  default =  "/vicon/mav_felipe/mav_felipe"/>
	<arg name = "airsim_world_frame"  default =  "world_enu"/>
	<!-- <arg name = "airsim_drone_frame"  default =  "/$(arg drone_name)"/> -->
	<!-- <arg name =  "actor_name" default = "CustomizationCharacter_2" /> -->
   	<!-- <arg name = "pose_topic_actor"  default =  "/actor_pose"/> -->


	<!-- <arg name = "mav_frame_id"  default =  "iris_fpv_cam__iris__base_link"/> -->
	<!-- <arg name = "mav_frame_id"  default =  "front_center_custom_body"/> -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="attach_enu_actor" args="0 0 0 1 0 0 0  $(arg actor_name)_ned /$(arg actor_name)_enu 100" /> -->
	<node pkg="tf" type="static_transform_publisher" name="map_matching" args="0 0 0 0 0 0 1   /$(arg airsim_world_frame) /map 100" />
 
    <!--tf from airsim to external pose topic (vicon). In GPS mode, we don't need external topic (vicon). So we just feed dummy pose -->	
    <node name="tf2pose" pkg="px4_code" type="tf2poseStamped_node" output="screen">   
	    <param name="map_frame" value="$(arg airsim_world_frame)"/>
        <param name="base_frame" value="map"/>
        <param name="output_pose_topic" value="$(arg pose_topic)"/>
	</node>

<!-- 
    <node name="tf2pose_actor" pkg="px4_code" type="tf2poseStamped_node" output="screen">   
	    <param name="map_frame" value="world_enu"/>
        <param name="base_frame" value="/$(arg actor_name)_enu"/>
        <param name="output_pose_topic" value="$(arg pose_topic_actor)"/>
	</node> -->
    


    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find px4_code)/rviz/airsim_default.rviz"/>
  
	
<!--airsim -->	
    <include file = "$(find px4_code)/launch/airsim/airsim_node.launch"/>
    <!-- <include file = "$(find px4_code)/launch/airsim/depth_to_pointcloud.launch"/> -->
 
<!--mav wrapper-->
    <node name="mav_wrapper_node" pkg="px4_code" type="mav_wrapper_node" output="screen">   
              
        <param name="world_frame_id" value="/map"/>
        <param name="external_pose_topic" value="$(arg pose_topic)"/>
        <param name="pose_on" value="true"/>            
    </node>
<!--mav wrapper-->
    <include file = "$(find mavros)/launch/px4.launch">
      <arg name="fcu_url" value="udp://:14550@127.0.0.1:18570"/> 
            <!-- <arg name="fcu_url" value="udp://:14540@127.0.0.1:14557"/>  -->

    </include>
   
   <!--mav gcs-->
       <node name="mav_gcs_node" pkg="px4_code" type="mav_gcs_node">
	           <!-- <arg name="planner_node_name" value="/planner_client"/>          -->
		</node>


</launch>
